src/audio/buffer.c

```diff
@@ -30,13 +30,6 @@ struct comp_buffer *buffer_alloc(uint32_t size, uint32_t caps, uint32_t align)
 
 	tr_dbg(&buffer_tr, "buffer_alloc()");
 
-	/* validate request */
-	if (size == 0 || size > HEAP_BUFFER_SIZE) {
```

Comment:
Doesn't size == 0 mean the topology is bad ?

Reply:
maybe let me keep the 'size==0' check.